### ISSUE TITLE
Does SQLFluff support MySQLs `DEFAULT CURRENT_TIMESTAMP`?

### ISSUE THREAD

**Issue by MontealegreLuis** [sqlfluff/sqlfluff#1526]
## Expected Behaviour

- SQLFLuff reporting issues on `CREATE TABLE` statements that include fields with `DEFAULT CURRENT_TIMESTAMP`

## Observed Behaviour

- SQLFluff reporting an unparsable section

## Steps to Reproduce

- Create SQL file `test.sql` with following content

```sql
CREATE TABLE a_table(
    created_date DATETIME DEFAULT CURRENT_TIMESTAMP NOT NULL
);
```
 
- Execute SQLFluff `lint`

```bash
sqlfluff lint --dialect=mysql test.sql
```

Find error message below

```
== [db/migrations/test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE a_table(\n created_date D...'
All Finished 📜 🎉!
```

Link to [SQLFluff online](https://online.sqlfluff.com/fluffed?sql=H4sIAH_tXWEC_3MOcnUMcVUIcXTycVVIjC9JTMpJ1eDlUgCC5KLUxJLUlPgUIKngAlQV4unrquDi6uYY6hOi4BwaFOTqFxIPEgwOcfQNUPDzD1HwC_Xx4eXStOblAgAXYkCLWQAAAA%3D%3D&dialect=mysql) example

## Dialect

MySQL

## Version

```
Python 3.9.7
pip 21.2.4 from /usr/local/lib/python3.9/site-packages/pip (python 3.9)
sqlfluff, version 0.6.8
```

## Configuration

```
[sqlfluff:rules:L004]
indent_unit = space

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
capitalisation_policy = upper

[sqlfluff:rules:L040]
capitalisation_policy = upper
```
